keras/engine/training_utils.py

```diff
@@ -322,6 +322,10 @@ def collect_metrics(metrics, output_names):
         return [copy.copy(metrics) for _ in output_names]
     elif isinstance(metrics, dict):
         nested_metrics = []
+        if not set(metrics.keys()).issubset(set(output_names)):
+            unknown_output_names = list(set(metrics.keys()) - set(output_names))
+            warnings.warn('Invalid layer name for metric computations: '
+                          '{}'.format(unknown_output_names))
```

Comment:
I would be faster for users to fix their code with the following message:

`Invalid output name for metric computation: 'incorrect_name_here', the output names of your model are 'output_names_here'`